web/profiles/joinup/joinup.behat.inc

```diff
@@ -68,6 +73,37 @@ class JoinupSubContext extends DrupalSubContextBase implements DrupalSubContextI
   }
 
   /**
+   * Creates and authenticates a user with the given og role(s).
+   *
+   * Multiple roles can be passed separated with comma.
+   *
+   * @Given I am logged in as a user with the :role role(s) of the :rdf_entity :rdf_entity_bundle
+   * @Given I am logged in as a/an :role of the :rdf_entity :rdf_entity_bundle
+   */
+  public function assertAuthenticatedByOgRole($role, $rdf_entity, $rdf_entity_bunlde) {
+    $entity = $this->getRdfEntityByLabel($rdf_entity, $rdf_entity_bunlde);
+    $roles = explode(',', $role);
+    $roles = array_map('trim', $roles);
+    $roles = $this->convertOgRoleNamesToIds($roles, $entity);
+
+    // Check if a user with this role is already logged in.
+    if (!$this->loggedInWithOgRole($roles, $entity)) {
```

Comment:
Ah OK this explains why the `loggedInWithOgRole()` returns TRUE|FALSE instead of throwing an exception.

So `loggedInWithOgRole()` is simply not meant as a standalone assertion but rather as a helper method for this particular assertion. Let's move it out of the trait then, so people don't use it in their tests.